scala/scala-impl/test/org/jetbrains/plugins/scala/base/libraryLoaders/SmartJDKLoader.scala

```diff
@@ -134,11 +126,21 @@ object SmartJDKLoader {
           .flatMap(findJDK)
         }
 
-  private def currentJava(versionMajor: String) =
-    Some(SystemInfo.JAVA_VERSION)
-      .filter(v => v.startsWith(s"1.$versionMajor") || v.startsWith(versionMajor))
-      .flatMap(_ => sys.props.get("java.home"))
-      .flatMap(d => findJDK(new File(d).getParentFile))
+  private def findJDK(dir: File) = {
+    val macDir = new File(dir, "/Contents/Home") // mac workaround
+    val candidates = List(macDir, dir, new File(dir, "/Home"))
+    candidates
+      .filter(_.isDirectory)
+      .find {
+        _
+          .listFiles()
+          .exists { bin =>
+            bin.getName == "bin" &&
+              bin.listFiles().exists(javac => javac.getName == "javac.exe" || javac.getName == "javac")
+          }
+      }.orElse(Some(dir))
+  }
+
```

Comment:
Can you please restore the order of the methods such that the diff is smaller? Thanks.

1. `findJkd`
2. `inJvm`
3. `currentJava`

Reply:
Hi Vasil,

Functions have been reordered as requested.

Thanks.